modules/dxp/apps/portal-workflow/portal-workflow-kaleo-designer-web-test/src/testIntegration/java/com/liferay/portal/workflow/kaleo/designer/web/internal/upgrade/v1_0_1/test/KaleoDefinitionVersionUpgradeProcessTest.java

```diff
@@ -0,0 +1,177 @@
+import com.liferay.portal.workflow.kaleo.service.KaleoDefinitionVersionLocalService;
+
+import java.sql.Connection;
+import java.sql.PreparedStatement;
+import java.sql.Timestamp;
+
+import org.junit.After;
+import org.junit.Assert;
+import org.junit.Before;
+import org.junit.ClassRule;
+import org.junit.Rule;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+
+/**
+ * @author Nathaly Gomes
+ */
+@RunWith(Arquillian.class)
+public class KaleoDefinitionVersionUpgradeProcessTest {
+
+	@ClassRule
+	@Rule
+	public static final AggregateTestRule aggregateTestRule =
+		new AggregateTestRule(
+			new LiferayIntegrationTestRule(),
+			PermissionCheckerMethodTestRule.INSTANCE);
+
+	@Before
+	public void setUp() throws Exception {
+		_db = DBManagerUtil.getDB();
+
+		_draftVersion = RandomTestUtil.randomInt();
+		_name = RandomTestUtil.randomString();
+		_version = RandomTestUtil.randomInt();
+
+		_addKaleoDefinition();
+		_addKaleoDraftDefinition();
+	}
+
+	@After
+	public void tearDown() throws Exception {
+		_db.runSQL("DROP_TABLE_IF_EXISTS(KaleoDraftDefinition)");
+	}
+
+	@Test
+	public void testUpgrade() throws PortalException {
+		KaleoDefinitionVersion kaleoDefinitionVersion =
+			_kaleoDefinitionVersionLocalService.fetchKaleoDefinitionVersion(
+				_kaleoDefinition.getCompanyId(), _kaleoDefinition.getName(),
+				_kaleoDefinition.getVersion() + StringPool.PERIOD + 0);
+
+		Assert.assertEquals(
+			kaleoDefinitionVersion.getKaleoDefinitionId(),
+			_kaleoDefinition.getKaleoDefinitionId());
+
+		Assert.assertEquals(
+			kaleoDefinitionVersion.getStatus(),
+			WorkflowConstants.STATUS_APPROVED);
+
+		UpgradeProcess upgradeProcess = UpgradeTestUtil.getUpgradeStep(
+			_upgradeStepRegistrator, _CLASS_NAME);
+
+		upgradeProcess.upgrade();
+
+		kaleoDefinitionVersion =
+			_kaleoDefinitionVersionLocalService.fetchKaleoDefinitionVersion(
+				_kaleoDefinition.getCompanyId(), _kaleoDefinition.getName(),
+				_version + StringPool.PERIOD + --_draftVersion);
+
+		Assert.assertEquals(
+			kaleoDefinitionVersion.getKaleoDefinitionId(),
+			_kaleoDefinition.getKaleoDefinitionId());
+
+		Assert.assertEquals(
+			kaleoDefinitionVersion.getStatus(), WorkflowConstants.STATUS_DRAFT);
+	}
+
+	private void _addKaleoDefinition() throws Exception {
+		_kaleoDefinition = _kaleoDefinitionLocalService.addKaleoDefinition(
+			null, _name, RandomTestUtil.randomString(),
+			RandomTestUtil.randomString(), null, StringPool.BLANK, 1,
+			ServiceContextTestUtil.getServiceContext());
+	}
+
+	private void _addKaleoDraftDefinition() throws Exception {
+		_db.runSQL(
+			StringBundler.concat(
+				"create table KaleoDraftDefinition (kaleoDraftDefinitionId ",
+				"LONG not null primary key,groupId LONG,companyId LONG,userId ",
+				"LONG,userName VARCHAR(75) null,createDate DATE null,",
+				"modifiedDate DATE null,name VARCHAR(75) null,title STRING ",
+				"null,content TEXT null,version INTEGER,draftVersion INTEGER)",
+				""));
```

Comment:
@vngoliveira is it possible to simplify by creating a table only with the necessary columns?